modules/@apostrophecms/page/index.js

```diff
@@ -178,8 +178,16 @@ module.exports = {
           const criteria = self.getIdCriteria(_id);
           const result = await self.getRestQuery(req).and(criteria).toObject();
 
-          if (self.options.cache && self.options.cache.api) {
+          if (self.options.cache && self.options.cache.api && self.options.cache.api.maxAge) {
             self.setMaxAge(req, self.options.cache.api.maxAge);
+            self.sendETag(req, result);
+
+            if (self.doesETagMatch(req, result)) {
+              req.res.status(304);
```

Comment:
As mentioned in https://github.com/apostrophecms/apostrophe/pull/3692/files#r838668454, we're not forced to set the `304` status as Express seems to automatically send it (using ETag and If-None-Match).
Should we keep sending this `304` manually here, to be explicit on why we're sending an empty response below?